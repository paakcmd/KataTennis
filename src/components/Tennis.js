import React from 'react';
import { connect } from 'react-redux';
import { scoreUpdater } from '../actions/score';
import { displayScore } from '../domain-logic/score-display';

export const Tennis = ({ scoreUpdater, tennis: { p1, p2 } }) =>
  <div id="wrapper" >
    <div id="row">
      <h1>Kata Tennis</h1>
      <div className="name-box">{ displayScore({ p1, p2 }) }</div>
      <div id="score-box1" className="score-box">{ p1 }</div>
      <div id="score-box2" className="score-box">{ p2 }</div>
      <button id="p1-scoreInc-button" onClick={() => scoreUpdater('p1')}>Increase P1 score</button>
      <button id="p2-scoreInc-button" onClick={() => scoreUpdater('p2')}>Increase P2 score</button>
    </div>
  </div>

export default connect( state => { return { tennis: state.tennis }}, { scoreUpdater })(Tennis);
