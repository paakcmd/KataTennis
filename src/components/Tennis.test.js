import React from 'react';
import { shallow, mount } from 'enzyme';
import { Tennis } from './Tennis';
import '../setupTests.js';
import '../tempPolyfills.js';

describe('Tennis', () => {
  const mockScoreUpdater = jest.fn()
  const props = { tennis: { p1: 0, p2: 0 }, scoreUpdater: mockScoreUpdater };
  const tennis = shallow(<Tennis {...props} />);

  it('renders properly', () => {
    expect(tennis).toMatchSnapshot();
  });

  it('displays name from props', () => {
    expect(tennis.find('.name-box').text()).toEqual('Love-ALL');
  });

  it('displays scores from props', () => {
    expect(tennis.find('#score-box1').text()).toEqual('0');
    expect(tennis.find('#score-box2').text()).toEqual('0');

  });

  describe('when user click on the buttons', () => {
    beforeEach(() => {
      tennis.find('#p1-scoreInc-button').simulate('click');
      tennis.find('#p2-scoreInc-button').simulate('click');
    });

    it('updates score on each of score boxes', () => {
      expect(mockScoreUpdater).toHaveBeenCalledWith('p1');
      expect(mockScoreUpdater).toHaveBeenCalledWith('p2');
    });
  });
});
