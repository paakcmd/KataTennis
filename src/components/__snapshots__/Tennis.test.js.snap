// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tennis renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    scoreUpdater={[Function]}
    tennis={
        Object {
            "p1": 0,
            "p2": 0,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        id="row"
>
        <h1>
                Kata Tennis
        </h1>
        <div
                className="name-box"
        >
                Love-ALL
        </div>
        <div
                className="score-box"
                id="score-box1"
        />
        <div
                className="score-box"
                id="score-box2"
        />
        <button
                id="p1-scoreInc-button"
                onClick={[Function]}
        >
                Increase P1 score
        </button>
        <button
                id="p2-scoreInc-button"
                onClick={[Function]}
        >
                Increase P2 score
        </button>
</div>,
      "id": "wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Kata Tennis
</h1>,
          <div
            className="name-box"
>
            Love-ALL
</div>,
          <div
            className="score-box"
            id="score-box1"
/>,
          <div
            className="score-box"
            id="score-box2"
/>,
          <button
            id="p1-scoreInc-button"
            onClick={[Function]}
>
            Increase P1 score
</button>,
          <button
            id="p2-scoreInc-button"
            onClick={[Function]}
>
            Increase P2 score
</button>,
        ],
        "id": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Kata Tennis",
          },
          "ref": null,
          "rendered": "Kata Tennis",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Love-ALL",
            "className": "name-box",
          },
          "ref": null,
          "rendered": "Love-ALL",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
            "className": "score-box",
            "id": "score-box1",
          },
          "ref": null,
          "rendered": 0,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
            "className": "score-box",
            "id": "score-box2",
          },
          "ref": null,
          "rendered": 0,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Increase P1 score",
            "id": "p1-scoreInc-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Increase P1 score",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Increase P2 score",
            "id": "p2-scoreInc-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Increase P2 score",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          id="row"
>
          <h1>
                    Kata Tennis
          </h1>
          <div
                    className="name-box"
          >
                    Love-ALL
          </div>
          <div
                    className="score-box"
                    id="score-box1"
          />
          <div
                    className="score-box"
                    id="score-box2"
          />
          <button
                    id="p1-scoreInc-button"
                    onClick={[Function]}
          >
                    Increase P1 score
          </button>
          <button
                    id="p2-scoreInc-button"
                    onClick={[Function]}
          >
                    Increase P2 score
          </button>
</div>,
        "id": "wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Kata Tennis
</h1>,
            <div
              className="name-box"
>
              Love-ALL
</div>,
            <div
              className="score-box"
              id="score-box1"
/>,
            <div
              className="score-box"
              id="score-box2"
/>,
            <button
              id="p1-scoreInc-button"
              onClick={[Function]}
>
              Increase P1 score
</button>,
            <button
              id="p2-scoreInc-button"
              onClick={[Function]}
>
              Increase P2 score
</button>,
          ],
          "id": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Kata Tennis",
            },
            "ref": null,
            "rendered": "Kata Tennis",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Love-ALL",
              "className": "name-box",
            },
            "ref": null,
            "rendered": "Love-ALL",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
              "className": "score-box",
              "id": "score-box1",
            },
            "ref": null,
            "rendered": 0,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
              "className": "score-box",
              "id": "score-box2",
            },
            "ref": null,
            "rendered": 0,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Increase P1 score",
              "id": "p1-scoreInc-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Increase P1 score",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Increase P2 score",
              "id": "p2-scoreInc-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Increase P2 score",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
